conans/client/installer.py

```diff
@@ -121,20 +142,30 @@ def _prepare_sources(self, conanfile, pref, package_layout, conanfile_path, sour
             logger.debug("BUILD: Copied to %s", build_folder)
             logger.debug("BUILD: Files copied %s", ",".join(os.listdir(build_folder)))
 
-    def _build(self, conanfile, pref):
+    def _build(self, conanfile, pref, node):
         # Read generators from conanfile and generate the needed files
         logger.info("GENERATORS: Writing generators")
         self._generator_manager.write_generators(conanfile, conanfile.build_folder, self._output)
 
         logger.info("TOOLCHAIN: Writing toolchain")
         write_toolchain(conanfile, conanfile.build_folder, self._output)
 
+        # I need the shims for the executables in the build-requires (only two-profiles approach)
+        if self._cache.config.shims_enabled:
+            logger.info("SHIMS: Writing shims for build-requires")
+            _, _, build_deps = _classify_dependencies(node)
```

Comment:
And if the refactor is done, this could be moved later, so ``classify_dependencies()`` is called just one? It would be a bit more natural.